slurm_ops_manager/slurm_ops_base.py

```diff
@@ -267,13 +270,104 @@ def upgrade(self):
         """Preform upgrade-charm operations."""
         raise Exception("Inheriting object needs to define this method.")
 
-    def setup_system(self):
-        """Preform the install and setup operations."""
+    def setup_slurm(self):
+        """Install and setup Slurm and its dependencies."""
         raise Exception("Inheriting object needs to define this method.")
 
+    @property
+    def nhc_version(self) -> str:
```

Comment:
is this used anywhere else, or can we make it a private property?

Reply:
`slurm_version` is also public, i followed that. I could make it private though.

Maybe we should leave the `foo_version` public so we can have some actions to retrieve the version of the components installed? slurm, nhc, infiniband, mpi, etc?

That would be cool to have:
```bash
$ juju run-action slurmd version
Omnivector Slurm Distribution version x.y.z
slurm 20.11.5
NHC 1.4.2-omni-1.0
Mellanox Infiniband 5.0.1
OpenMPI 4.0.5
```